### ISSUE TITLE
Why does it parse comments in environment variables

### ISSUE THREAD

**Issue by mmsaki** [theskumar/python-dotenv#515]
# Example

This is my set up for `.env` file
```toml
VARIABLE="xxxxxxx" # this is a comment
```

## Usage with `dotenv`

```py
import os
from dotenv import load_dotenv
load_dotenv()

variable = os.getenv("VARIABLE")
print(variable)
```

## Issue

Currently the usage of dotenv produces the following output:

```sh
variable = '"xxxxxxx" # this is a comment'
```

## Expected usage

I was expecting the `dotenv` package to parse my variable without including any inline comments

```py
variable = "xxxxxxx"
```

**Comment by mmsaki** [comment]
Also the dotenv package is not watching for real time changes in the `.env` file. Why is this happening? This is so bad security wise. Is the file being cached?

**Comment by Bajron** [comment]
I cannot reproduce your issue with the provided input.
There is however a behavior in this package that you might have encountered with slightly different input.

If your .env is (note the '#' glued to the first word):
```
VARIABLE=xxxxxxx# this is a comment
```
Then indeed the '#' does not behave as an inline comment indicator. This behavior is kind of consistent with bash.

If you `source .env` in bash you get `bash: this: command not found`
If your .env is:
```
VARIABLE=xxxxxxx#this_is_NOT_a_comment
```
and you source it in bash, you will see the VARIABLE is set to "xxxxxxx#this_is_NOT_a_comment".
So you need a space before the '#' to make it a comment in bash.

The other issue you observe might be the fact, that `load_dotenv` does not override existing variables by default. You can control this behavior with the named argument `override`. For example `load_dotenv(override=True)`.

**Comment by th0rgall** [comment]
I had an issue that looked similar, but it had nothing to do with `python-dotenv`, but rather with behavior of VSCode that I didn't understand. I'm leaving this note if anyone else finds their way here, because I think it's highly likely this was @mmsaki's underlying issue.

It seems the built-in Python extension of VSCode has a `python.envFile` setting that is by default set to `${workspaceFolder}/.env` ([source](https://code.visualstudio.com/docs/python/environments#_environment-variable-definitions-file)). Every time you open a terminal, VSCode loads your `.env` file into it, in its own idiosyncratic way (see below).

If you aren't aware of this default behavior, like I was, this can lead to very unexpected results. For example, it could explain the "caching" behavior that @mmsaki observed, which was also my issue. Here's how to reproduce it:

1. Have an `.env` file in a Python project with some env vars.
1. Open a new VSCode terminal.
2. Change, add or remove variables in your `.env` file.
3. Call a Python program that uses `python-dotenv`'s `load_dotenv()` in the same terminal (without `override=True`)
4. Observe: the initial environment variables & their values values from step (1), as sourced by VSCode, will have overridden the actual state of your `.env`. You're working with old state created before the program executed.

Additionally, I've tested the initial reproduction case described by @mmsaki, which also matches in behavior:

1. Add `VARIABLE="xxxxxxx" # this is a comment` to your `.env`
2. Have the`python.envFile` workspace setting set to the default value
2. Open a new VSCode terminal
3. Run `echo $VARIABLE`
4. Output: `"xxxxxxx" # this is a comment`

This is idiosyncratic, since it is different from how `zsh` and `bash` source such a variable.

**Comment by mmsaki** [comment]
@th0rgall yes, this was exactly the behavior I was hoping to highlight when I opened this issue. Thanks for sharing your discovery and uncovering the unexpected behaviors are coming from the built-in Python extension in VSCode. Thanks for your efforts, I am sure it'll help a lot of people that come across this issue again.

**Comment by branchard** [comment]
@th0rgall Or simply do: 
```py
import os
from dotenv import load_dotenv

for name in os.environ:
    os.environ.pop(name)

load_dotenv()

print(os.environ)
```

**Comment by branchard** [comment]
The easiest way to fix this is to use the `override` argument of `load_dotenv` to override the variables that are loaded by Visual Studio Code.
```py
import os
from dotenv import load_dotenv

load_dotenv(override=True)
print(os.environ)
```
